Controller/Adminhtml/Giftcard/Save.php

```diff
@@ -20,21 +20,55 @@
 
 namespace Buckaroo\Magento2\Controller\Adminhtml\Giftcard;
 
+use Magento\Framework\App\Filesystem\DirectoryList;
+
 class Save extends \Buckaroo\Magento2\Controller\Adminhtml\Giftcard\Index
 {
     public function execute()
     {
         $isPost = $this->getRequest()->getPost();
 
         if ($isPost) {
+
             $giftcardModel = $this->giftcardFactory->create();
-            $giftcardId = $this->getRequest()->getParam('entity_id');
+            $giftcardId    = $this->getRequest()->getParam('entity_id');
 
             if ($giftcardId) {
                 $giftcardModel->load($giftcardId);
             }
 
             $formData = $this->getRequest()->getParam('giftcard');
+
+            if ((isset($_FILES['logo']['name'])) && ($_FILES['logo']['name'] != '') && (!isset($formData['logo']['delete']))) {
```

Comment:
$this->getRequest()->getFiles()
will it work in more magento native way ?

Reply:
fixed